libs/d3d12core/main.c

```diff
@@ -198,11 +201,17 @@ static VkPhysicalDevice d3d12_find_physical_device(struct vkd3d_instance *instan
 
         if (id_properties.deviceLUIDValid && !memcmp(id_properties.deviceLUID, &adapter_desc->AdapterLuid, VK_LUID_SIZE))
         {
-            vk_physical_device = vk_physical_devices[i];
-            break;
+            if(properties2.properties.deviceType == VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU && !dgpu_device)
+                dgpu_device = vk_physical_devices[i];
+            if(properties2.properties.deviceType == VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU && !igpu_device)
+                igpu_device = vk_physical_devices[i];
```

Comment:
I'd rather just do something like

```c
vk_physical_device = vk_physical_devices[i];

if (properties2.properties.deviceID == adapter_desc->deviceId &&
    properties2.properties.vendorID == adapter_desc->vendorId)
  break;
```

Reply:
See comment. Multiple adapters sharing the same LUID shouldn't be happening in the first place, but instead of adding some weird heuristic we should make an effort to actually match the device that the app wants to use by checking the remaining adapter properties.

Reply:
I was thinking about this, but would it work when dxgi "fakes" a different vendor id on nvidia?

Reply:
That's a problem, but at least there are ways to work around that from the Proton/DXVK side.

Reply:
I guess we could also look at the device name as a last resort. That can technically be faked too, but no part in the stack is going to do that by default and DXVK just forwards the raw Vulkan device name anyway.